ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,303 @@
+
+
+def use_external_elb(**_):
+
+    utils.use_external_resource(ctx.node.properties)
+
+
+@operation
+def delete_elb(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        utils.unassign_runtime_property_from_resource(
+            constants.EXTERNAL_RESOURCE_ID, ctx.instance)
+        return
+
+    ctx.logger.info('Attempting to delete Load Balancer.')
+
+    elb_name = ctx.node.properties['elb_name']
+
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.delete()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer {0} not deleted. '
+                                  .format(str(e)))
+
+    ctx.logger.info(
+        'Load Balancer {0} deleted. '
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return
+
+
+def _create_health_check(user_health_check):
+
+    health_check = {
+        'interval': constants.HEALTH_CHECK_INTERVAL,
+        'healthy_threshold': constants.HEALTH_CHECK_HEALTHY_THRESHOLD,
+        'timeout': constants.HEALTH_CHECK_TIMEOUT,
+        'unhealthy_threshold': constants.HEALTH_CHECK_UNHEALTHY_THRESHOLD
+    }
+
+    health_check.update(user_health_check)
+
+    health_check = HealthCheck(**health_check)
+    return health_check
+
+
+def _create_elb_params():
+    params_dict = {'listeners': ctx.node.properties['listeners'],
+                   'name': ctx.node.properties['elb_name'],
+                   'zones': ctx.node.properties['zones']}
+    if 'security_groups' in ctx.node.properties.keys():
+        params_dict['security_groups'] = ctx.node.properties['security_groups']
+    return params_dict
+
+
+def _create_elb():
+
+    ctx.logger.info('Attempting to Create Load Balancer.')
+
+    params_dict = _create_elb_params()
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
+        lb = elb_client.create_load_balancer(**params_dict)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer not created '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info('Load Balancer Created. ')
+
+    utils.set_external_resource_id(value=params_dict['name'],
+                                   ctx_instance=ctx.instance,
+                                   external=False)
+
+    ctx.instance.runtime_properties['elb_name'] = params_dict['name']
+
+    if not lb:
+        raise NonRecoverableError('Load Balancer not created')
+
+    return lb
+
+
+def _get_instance_list():
+
+    ctx.logger.info('Attempting to get Load Balancer Instance List.')
+
+    elb_name = ctx.node.properties['elb_name']
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        list_of_instances = lb.instances
```

Comment:
when in ipython, I tried to call lb.instances before and after adding an instance and got real results, which means it is communicating externally.